src/deadline/client/api/_session.py

```diff
@@ -67,36 +60,32 @@ def get_boto3_session(
     if profile_name in ("(default)", "default", ""):
         profile_name = None
 
-    # If a config was provided, don't use the Session caching mechanism.
-    if config:
-        return boto3.Session(profile_name=profile_name)
-
```

Comment:
Unclear why this is hear. It appears to be a mistake because `config` not part of the created session. The comment doesn't give more context. It was in this repo's initial commit so there's also no info in the git history. The new `lru_cache` is keyed off of profile name instead which is actually used in the session creation and also conveniently hashable.

Reply:
Huge thanks to @epmog for tracking this down in the pre-release repo. This code is a holdover from when we passed `config` to the client constructor, so it'd make sense to not reuse a session from a different config. Since the original commit, we switched to only passing the profile from `config` to the client constructor, so now we can reuse/cache a session if the profile name hasn't changed.

We can also see where the profile name is extracted from `config` here before it's passed to the client constructor: https://github.com/aws-deadline/deadline-cloud/blob/mainline/src/deadline/client/api/_session.py#L63